apps/cms/src/collections/Products.ts

```diff
@@ -0,0 +1,126 @@
+import { CollectionConfig } from "payload/types";
+
+/** Products collection stores merch products offerings. */
+export const Products: CollectionConfig = {
+  slug: "products",
+  admin: {
+    description: "Merchandise products offerings",
+  },
+  fields: [
+    // by default, payload generates an 'id' field each order automatically
+    {
+      name: "name",
+      type: "text",
+      required: true,
+    },
+    {
+      name: "colors",
+      type: "select",
+      hasMany: true,
+      options: [
+        {
+          label: "Black",
+          value: "black",
+        },
+        {
+          label: "White",
+          value: "white",
+        },
+        {
+          label: "Blue",
+          value: "blue",
+        },
+      ],
+    },
```

Comment:
It might make more sense for the type to be "text" with `hasMany` so we allow the business team to add colours as necessary. The same applies to `sizes`, `category` and `stock`.

Reply:
We are on Payload CMS 1.8.3, which [does not have](https://github.com/payloadcms/payload/blob/51fd1db4eb45ff4d03be043d0b4af59e7fbe285e/docs/fields/text.mdx) `hasMany` on the `text.

Two options:
1. Upgrade Payload CMS to >2.0, potentially having to stomach breaking changes / large rewrites in the port.
2. Add a `colors`, `sizes`, `category` & `stock` collections which can be amended dynamically by the business team and then use the `relationship` type from the `product` collection to link to color, sizes etc. However, this normalises the data model which is contradicts the purpose of using a document data model in MongoDB in the first place: colocating data accessed together. Normalisation is more suitable to the generic SQL relational DBs like PostgreSQL.
3. Do nothing, consider this a premature optimization: one can quickly add options if necessary.

Reply:
Does it make sense to have `stock`  + `colors` & `sizes`? There seems to be data duplication here. We can have 1 `stock` field instead.

Reply:
I believe this might be mixing up different concerns and we can keep both of them for now. Some color/size combo may be valid but do not have any stock (for which the `stock` field may not contain the associated keys).

Reply:
Updated to `hasMany`.